File: SatinalmaPro.Android/app/src/main/java/com/satinalmapro/shared/filter/ProcurementTab.kt
package com.satinalmapro.shared.filter

enum class ProcurementTab {
    PENDING_DRAFT,
    MANAGEMENT_APPROVAL,
    QUOTE_REQUESTED,
    QUOTE_REVIEW,
    APPROVED_ORDERS,
    REJECTED,
    HISTORY_REPORTS,
    STOCK_MOVEMENTS,
    STOCK_STATUS;

    val title: String
        get() = when (this) {
            PENDING_DRAFT -> "Beklemede / Taslak"
            MANAGEMENT_APPROVAL -> "Yönetim Onayda"
            QUOTE_REQUESTED -> "Teklif İstendi"
            QUOTE_REVIEW -> "Teklif İnceleme"
            APPROVED_ORDERS -> "Onaylananlar / Sipariş"
            REJECTED -> "Reddedilenler"
            HISTORY_REPORTS -> "Geçmiş Raporlar"
            STOCK_MOVEMENTS -> "Stok / Hareketler"
            STOCK_STATUS -> "Stok Durumu"
        }

    val routeKey: String
        get() = when (this) {
            PENDING_DRAFT -> "pending-draft"
            MANAGEMENT_APPROVAL -> "management-approval"
            QUOTE_REQUESTED -> "quote-requested"
            QUOTE_REVIEW -> "quote-review"
            APPROVED_ORDERS -> "approved-orders"
            REJECTED -> "rejected"
            HISTORY_REPORTS -> "history-reports"
            STOCK_MOVEMENTS -> "stock-movements"
            STOCK_STATUS -> "stock-status"
        }

    val isProcurementCollection: Boolean
        get() = this != STOCK_MOVEMENTS && this != STOCK_STATUS
}

File: SatinalmaPro.Shared/Procurement/ProcurementTab.cs
namespace SatinalmaPro.Shared.Procurement;

/// <summary>Satınalma modülü ana sekmeleri — Android ve masaüstü ortak.</summary>
public enum ProcurementTab
{
    PendingDraft,
    ManagementApproval,
    QuoteRequested,
    QuoteReview,
    ApprovedOrders,
    Rejected,
    HistoryReports,
    StockMovements,
    StockStatus
}

public static class ProcurementTabExtensions
{
    public static string Title(this ProcurementTab tab) => tab switch
    {
        ProcurementTab.PendingDraft => "Beklemede / Taslak",
        ProcurementTab.ManagementApproval => "Yönetim Onayda",
        ProcurementTab.QuoteRequested => "Teklif İstendi",
        ProcurementTab.QuoteReview => "Teklif İnceleme",
        ProcurementTab.ApprovedOrders => "Onaylananlar / Sipariş",
        ProcurementTab.Rejected => "Reddedilenler",
        ProcurementTab.HistoryReports => "Geçmiş Raporlar",
        ProcurementTab.StockMovements => "Stok / Hareketler",
        ProcurementTab.StockStatus => "Stok Durumu",
        _ => tab.ToString()
    };

    public static string RouteKey(this ProcurementTab tab) => tab switch
    {
        ProcurementTab.PendingDraft => "pending-draft",
        ProcurementTab.ManagementApproval => "management-approval",
        ProcurementTab.QuoteRequested => "quote-requested",
        ProcurementTab.QuoteReview => "quote-review",
        ProcurementTab.ApprovedOrders => "approved-orders",
        ProcurementTab.Rejected => "rejected",
        ProcurementTab.HistoryReports => "history-reports",
        ProcurementTab.StockMovements => "stock-movements",
        ProcurementTab.StockStatus => "stock-status",
        _ => "dashboard"
    };

    public static bool IsProcurementCollection(this ProcurementTab tab) =>
        tab is not (ProcurementTab.StockMovements or ProcurementTab.StockStatus);
}
